### DISCUSSION TITLE
Options with icons with single select

### DISCUSSION THREAD

**Discussion by alexnguyennz** [csandman/chakra-react-select#253] [Q&A] [answered]
Hi is there a version of [this example](https://codesandbox.io/s/chakra-react-select-custom-icon-components-typescript-odi90k?file=/app.tsx) but for single selection instead of multi (so the icon is displayed with the selected option)? 

I've tried something like:

```tsx
<Select
  value={{
    icon: <SomeComponent />, // similar to customComponent, render icon + label
    value: optionValue,
    label: (<SomeComponent />) as unknown as string 
  }} 
/>
```

Which works visually (making the current selection look just like the options) but it's not ideal.

Thanks.

**Comment by csandman** [comment, marked as answer]
Sorry for the super late reply, just started a new job and I've been very busy!

This isn't that hard if you make a custom `SingleValue` component (in the same way the example you linked does the `MultiValue`):

```tsx
const customComponents = {
  Option: ({ children, ...props }) => (
    <chakraComponents.Option {...props}>
      <Icon
        as={props.data.Icon}
        color={props.data.iconColor}
        mr={2}
        h={5}
        w={5}
      />
      {children}
    </chakraComponents.Option>
  ),
  SingleValue: ({ children, ...props }) => (
    <chakraComponents.SingleValue {...props}>
      <Flex align="center">
        <Icon
          as={props.data.Icon}
          color={props.data.iconColor}
          mr={2}
          h={5}
          w={5}
        />
        {children}
      </Flex>
    </chakraComponents.SingleValue>
  )
};
```

CodeSandbox: https://codesandbox.io/s/4gsnly?file=/app.tsx

**Reply by alexnguyennz** [reply]
No worries - congrats on your new job! 

That solution works great, thank you.

**Reply by csandman** [reply]
> congrats on your new job!

Thanks a bunch! glad it worked out for you!
